core/src/main/kotlin/org/eln2/sim/electrical/mna/Node.kt

```diff
@@ -1,18 +1,34 @@
 package org.eln2.sim.electrical.mna
 
 import org.eln2.debug.dprintln
+import org.eln2.sim.electrical.mna.component.*
+import org.eln2.space.Set
 
+/**
+ * A "node", in MNA; the non-resistive connections between [Component]s.
+ *
+ * Aside from identifying [Component]s' connections, the Nodes' potentials (relative to [Circuit.ground]) are computed as a result of the MNA algorithm.
+ */
 open class Node(var circuit: Circuit) : IDetail {
+	/**
+	 * The potential of this node, in Volts, relative to ground (as defined by the [Circuit]); an output of the simulation.
+	 */
 	open var potential: Double = 0.0
+	/**
+	 * The index of this node into the [Circuit]'s matrices and vectors.
+	 */
 	open var index: Int = -1  // Assigned by Circuit
+	/**
+	 * True if this node is ground (defined to have zero potential).
+	 */
 	open val isGround = false
 	var name = "node"
```

Comment:
Is this really something that can change after construction?

Reply:
It's a val, so I assume not? :)

Reply:
No, the name.

Reply:
The name `isGround` is problematic?

It's somewhat a shortcut to `node is GroundNode`, I suppose; I guess it could be replaced there if that test is faster. This current design affords some unneeded flexibility, now that the API is heading toward stable...

Reply:
Er.

What I mean is, can the value of `var name` actually change? Or should it be a val?

Reply:
...oh, the field `name`... GitHub's UI for annotating comments leaves some to be desired. I, uh, deleted isGround in the newest tip anyway; d'ya think that's something I should reintroduce?

The name is... a good question! It probably should be a val, and honestly might even best be static. It was a part of the code that I believe I inherited from @jrddunbr and didn't touch; I don't know what his plan was for it. In particular, it's too vague for me to tell myself...

Reply:
I'm not convinced @Baughn is aware of how to tag the lines properly. I have the same issue; we should compare notes.

The old code had a "name" variable in the SubSystem implementation. This was handy because 2hen you dumped out a circuit, you had an inkling of an idea of what it was by the name. I'd like to expand on this so that components as part of a particular device (macerator, cable, battery,etc) are visible since they will be an abstraction layer above the MNA.

We could go about it a different way, I just found having a name is nice. I'd imagine things like "battery xxyyzz positive electrical load" (hopefully we rename the electrical load component by the way).

Reply:
okay, so name is a variable intended by use for a higher layer; it can definitely stay for that purpose :)

Should I revert the "removing `isGround`" change?

Reply:
Copying intents from Discord: yes, revert the change and apply changes for the `name` field.